accelerator/multi_region_modules/hubNetworking.bicep

```diff
@@ -0,0 +1,1164 @@
+metadata name = 'ALZ Bicep - Hub Networking Module'
+metadata description = 'ALZ Bicep Module used to set up Hub Networking'
+
+type subnetOptionsType = ({
+  @description('Name of subnet.')
+  name: string
+
+  @description('IP-address range for subnet.')
+  ipAddressRange: string
+
+  @description('Id of Network Security Group to associate with subnet.')
+  networkSecurityGroupId: string?
+
+  @description('Id of Route Table to associate with subnet.')
+  routeTableId: string?
+
+  @description('Name of the delegation to create for the subnet.')
+  delegation: string?
+})[]
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. The lock settings of the service.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')
+
+  @description('Optional. Notes about this lock.')
+  notes: string?
+}
+
+@sys.description('The Azure Region to deploy the resources into.')
+param parLocation string = resourceGroup().location
+
+param parSecondaryLocation string?
+
+@sys.description('Prefix value which will be prepended to all resource names.')
+param parCompanyPrefix string = 'alz'
+
+@sys.description('''Global Resource Lock Configuration used for all resources deployed in this module.
+
+- `kind` - The lock settings of the service which can be CanNotDelete, ReadOnly, or None.
+- `notes` - Notes about this lock.
+
+''')
+param parGlobalResourceLock lockType = {
+  kind: 'None'
+  notes: 'This lock was created by the ALZ Bicep Hub Networking Module.'
+}
+
+
```

Comment:
We would maybe need to add a parameter for the primaryHubVnetId and call the peering module from here to do the peering